lib/graphql/resultPage.ts

```diff
@@ -0,0 +1,20 @@
+import { PageInfo } from "./schema";
+import { IFetch } from "./interfaces";
+
+export class ResultPage {
+  items: Array<Object>;
+  nextPage?: () => Promise<ResultPage>;
+  previousPage?: () => Promise<ResultPage>;
+
+  constructor(model: IFetch, items: Array<Object>, pageInfo: PageInfo) {
+    this.items = items;
```

Comment:
Any chance we can use generics here? 
```
export class ResultPage<T> {
  items: Array<T>;
  nextPage?: () => Promise<ResultPage<T>>;
  previousPage?: () => Promise<ResultPage<T>>;

  constructor(model: IFetch, items: Array<T>, pageInfo: PageInfo) {
```

then later in `transactions.ts` sth. like

```
export class Transaction extends Model {
   async fetch(args?: FetchOptions): Promise<ResultPage<Transaction>> {
```